kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py

```diff
@@ -26,35 +31,48 @@ def __init__(self, config_path):
         """
         Construct a DatasetIngestConfig object from a config file
 
-        :param config_filepath: Path to the data ingest config file
+        :param config_path: Path to the data ingest config file
         """
-        # Is config path a dir or file
+        # If given config path is a dir, create default config file path
         config_path = os.path.abspath(os.path.expanduser(config_path))
         if os.path.isdir(config_path):
-            # If dir, create default config file path
             config_path = os.path.join(
                 config_path,
-                DATASET_INGEST_CONFIG_DEFAULT_FILENAME)
-
-        super().__init__(config_path, schema_path=DATA_INGEST_SCHEMA_PATH)
-
-        # Directory containing the extract config files
-        self.extract_config_dir = os.path.join(
-            os.path.dirname(self.config_filepath),
-            self.contents.get('extract_config_dir'))
-        self.extract_config_paths = [os.path.join(self.extract_config_dir,
-                                                  filename)
-                                     for filename in os.listdir(
-                                     self.extract_config_dir)
-                                     if filename.endswith('.py')]
+                DATASET_INGEST_CONFIG_DEFAULT_FILENAME
+            )
+
+        super().__init__(config_path)
+
+        # Validate
+        try:
+            self.extract_config_dir
+            assert_safe_type(self.extract_config_dir, str)
+
+            # Directory containing the extract config files
+            self.extract_config_dir = os.path.join(
+                os.path.dirname(self.config_filepath),
+                self.extract_config_dir
+            )
+            self.extract_config_paths = [
```

Comment:
This should just be validation. We should separate the setting of config variables or any other business logic from validation logic.